config/application.rb

```diff
@@ -200,10 +200,15 @@ class Application < Rails::Application
     # Catching Invalid JSON Parse Errors with Rack Middleware
     config.middleware.use Foreman::Middleware::CatchJsonParseErrors
 
-    # When operating behind a reverse proxy, this provides a valid remote ip
-    if SETTINGS[:trusted_proxies]
-      config.middleware.use ActionDispatch::RemoteIp, custom_proxies: SETTINGS[:trusted_proxies]
-    end
+    # When operating behind a reverse proxy, this provides a valid remote ip in request.remote_ip
+    # the middleware is enabled by default however we minimize the trusted proxies list
+    #
+    # the trusted proxies are removed from remote IP candidates, since we know they are proxies and not clients
+    # in the default deployment, this is typically an Apache
+    #
+    # currently the requests with untrusted IPs in X_FORWARDED_FOR are taken into the consideration,
+    # the spoof detection based on HTTP_CLIENT_IP header performed by the middleware does not work
+    config.action_dispatch.trusted_proxies = %w(127.0.0.1/8 ::1).map { |proxy| IPAddr.new(proxy) }
```

Comment:
I still think a `SETTINGS[:trusted_proxies]` is good to have. It allows overriding via YAML if needed. It can default to localhost only (as you have it now).

Reply:
then we need the installer patch which I would prefer to avoid for the stable branch